### ISSUE TITLE
Allow a rule flag to enable rules that have been disabled globally.

### ISSUE THREAD

**Issue by kuwerty** [palantir/tslint#3032]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.3.4
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
/* tslint:enable:object-literal-sort-keys */
const range = {
   min: 5,
   middle: 10,  // out of order, report an error?
   max: 20,
};
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "jsRules": {},
    "rules": {
        "no-var-keyword": true,
        "object-literal-sort-keys": false
    },
    "rulesDirectory": []
}
```

#### Actual behavior
No error is reported at line 3 for the out of order key. The rule is disabled globally and the tslint:enable:object-literal-sort-keys has no effect.

This follows the rule flag docs that state: 'Enabling a rule that is not present or disabled in tslint.json has also no effect'.

#### Expected behavior
Suppose we have a large code base and don't want to convert the entire thing (all at once) to support one specific rule.  Ideally I would keep that rule disabled in the tslint.json. As files are fixed to support a new rule, enable a rule per-file.

Is there any workflow that supports this in tslint at the moment?

**Comment by phiggins** [comment]
I came here to post the same thing as a bug report. Based on that line in the documentation, I thought this should work. To clarify things, I'd rephrase this line:

> Enabling a rule that is not present or disabled in tslint.json has also no effect.

to something like:

> Enabling a rule that is not present or is disabled in tslint.json also has no effect.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
